### PR TITLE
topsql: add scraper tests

### PR DESCRIPTION
**Pull request by zhongzc**
Signed-off-by: Zhenchi <zhongzc_arch@outlook.com>

### CODE REVIEW

**Comment by zhongzc** [comment]
@mornyx Could you please take a look? I think you're a test master.

**Comment by mornyx** [comment]
Rest LGTM.

**Review by crazycs520 — APPROVED** [review]

File: component/topsql/mock/pubsub/tidb.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func (s *MockTiDBPubSub) Listen() (ip string, port uint, err error) {
+	s.listener, err = net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return
+	}
+
+	s.server = grpc.NewServer(
+		grpc.KeepaliveParams(keepalive.ServerParameters{
```
[the remark this answers is not in the document]
Comment (zhongzc) [reply component/topsql/mock/pubsub/tidb.go:44]: Yes

File: component/topsql/mock/pubsub/tikv.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func (s *MockTiKVPubSub) Listen() (ip string, port uint, err error) {
+	s.listener, err = net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return
+	}
+
+	s.server = grpc.NewServer(
+		grpc.KeepaliveParams(keepalive.ServerParameters{
```
[the remark this answers is not in the document]
Comment (zhongzc) [reply component/topsql/mock/pubsub/tikv.go:44]: Yes

**Pull request merged**